src/input/type_base.hh

```diff
@@ -360,6 +360,11 @@ public:
     : lower_bound_(lower_bound), upper_bound_(upper_bound)
     {}
 
+    template <class T>
```

Comment:
Finally we don't want this, since it could make platform dependent input tree.. Delete.